src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py

```diff
@@ -0,0 +1,217 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long, too-few-public-methods
+
+import json
+
+from knack.util import CLIError
+from knack.log import get_logger
+
+from ._constants import CompareFields, JsonDiff, AppServiceConstants
+from ._featuremodels import (map_keyvalue_to_featureflag, custom_serialize_conditions, is_feature_flag)
+from ._utils import is_json_content_type
+
+logger = get_logger(__name__)
+
+
+# Utility class used in diff-ing key-values to find added, deleted and updated key-values.
+class KVComparer:
+    def __init__(self, src_kvs, compare_fields, preserve_labels=True, label=None, content_type=None):
+        self._compare_fields = compare_fields
+        self._src_kvs = src_kvs
+
+        for kv in self._src_kvs:
+
+            if content_type and not is_feature_flag(kv):
+                kv.content_type = content_type
+
+            if not preserve_labels:
+                kv.label = label
+
+    # Returns a diff in the form {"add": List[KeyValue], "delete": List[KeyValue], "update": List[{"new": KeyValue, "old": KeyValue}]}
+    def compare(self, dest_kvs, strict=False):
+        if not dest_kvs:
+            return {JsonDiff.ADD: self._src_kvs}
+
+        if not strict and not self._src_kvs:
+            return {}
+
+        dest_kv_lookup = self._build_lookup(dest_kvs)
+
+        kv_diff = {}
+        kv_diff[JsonDiff.ADD] = []
+        kv_diff[JsonDiff.DELETE] = []
+        kv_diff[JsonDiff.UPDATE] = []
+
+        for kv in self._src_kvs:
+            lookup_key = (kv.key, kv.label or "")
+            if not dest_kv_lookup.get(lookup_key, None):
+                kv_diff[JsonDiff.ADD].append(kv)
+
+            else:
+                if not self._kv_equals(kv, dest_kv_lookup[lookup_key], self._compare_fields):
+                    kv_diff[JsonDiff.UPDATE].append({"new": kv, "old": dest_kv_lookup[lookup_key]})
+
+                del dest_kv_lookup[lookup_key]
+
+        if strict:
+            kv_diff[JsonDiff.DELETE].extend(dest_kv_lookup.values())
+
+        return kv_diff
+
+    @staticmethod
+    def _build_lookup(kvs):
+        if not kvs:
+            return {}
+
+        lookup = {}
+        for kv in kvs:
+            lookup[(kv.key, kv.label or "")] = kv
+
+        return lookup
+
+    @staticmethod
+    def _try_parse_json_value(value_string):
+        try:
+            return json.loads(value_string)
+
+        except Exception:  # pylint: disable=broad-except
+            return value_string
```

Comment:
Catch specific exceptions that `json.loads` might throw

Reply:
This has been updated, thanks!